### PR TITLE
Link test binaries dynamically for coverage builds

### PR DESCRIPTION
**Pull request by yanavlasov**
Commit Message:
Dynamic linking prevents build failure due to size of coverage instrumentation.

Additional Description:
This PR changed two known problematic binaries.

Risk Level: Low
Testing: Unit Tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: test/integration/run_envoy_test.sh [hunk 1]
```diff
@@ -52,6 +52,6 @@ expect_ok -c "${TEST_SRCDIR}/envoy/test/config/integration/empty.yaml" --mode va
 
 start_test "Launching envoy with empty config."
 run_in_background_saving_pid "${ENVOY_BIN}" -c "${TEST_SRCDIR}/envoy/test/config/integration/empty.yaml" --use-dynamic-base-id
-sleep 3
+sleep 120
```
Comment (RyanTheOptimist) [inline test/integration/run_envoy_test.sh:55]: Is this change intentional?

**Review by RyanTheOptimist — APPROVED** [review]
LGTM, modulo one question about an increased timeout.

File: test/integration/run_envoy_test.sh (hunk 1 above)
Comment (yanavlasov) [reply test/integration/run_envoy_test.sh:55]: Yes, otherwise the test fails in coverage. It takes over 1 minute in my tests to load coverage instrumented dynamically linked binary.
